graphula-core/src/Graphula/Internal.hs

```diff
@@ -51,6 +51,11 @@ class GHasDependenciesRecursive fieldsProxy node deps where
   genericDependsOnRecursive :: fieldsProxy -> node -> deps -> node
 
 -- This instance head only matches EoT representations of
+-- datatypes with no constructors. Ignore dependencies.
+instance GHasDependencies (Proxy nodeTy) (Proxy depsTy) Void deps where
```

Comment:
Talked offline. We decided that the improved errors trump the mild increase in complexity of these overlapping instances. :shipit: